### PR TITLE
[CH9786] Implement recommendation methods

### PR DESCRIPTION
**Pull request by sblaurock**
Implements method to retrieve recommendations from C.IO endpoint.

+ Tests
+ Response model

```
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running io.constructor.client.ConstructorIOBasicTest
Tests run: 19, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.903 sec
Running io.constructor.client.ModelsTest
Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 sec
Running io.constructor.client.ConstructorItemTest
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 sec
Running io.constructor.client.SearchRequestTest
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.003 sec
Running io.constructor.client.ConstructorIOSearchTest
Tests run: 12, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.872 sec
Running io.constructor.client.NaturalLanguageSearchRequestTest
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 sec
Running io.constructor.client.ConstructorIOAutocompleteTest
Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.189 sec
Running io.constructor.client.UserInfoTest
Tests run: 11, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 sec
Running io.constructor.client.ConstructorIOItemsTest
Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 8.324 sec
Running io.constructor.client.RecommendationsRequestTest
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 sec
Running io.constructor.client.ConstructorIORecommendationsTest
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.143 sec

Results :

Tests run: 82, Failures: 0, Errors: 0, Skipped: 0
```

Screen Shot 2020-05-12 at 5 44 10 PM

### CODE REVIEW

File: constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java [hunk 1]
```diff
@@ -530,6 +531,66 @@ public String naturalLanguageSearchAsJSON(NaturalLanguageSearchRequest req, User
...
+     * Queries the search service to retrieve recommendations.
+     *
+     * Note that if you're making a search service on a website, you should definitely use our javascript client instead of doing it server-side!
+     * That's important. That will be a solid latency difference.
+     *
+     * @param req the natural language search request
+     * @param userInfo optional information about the user
+     * @return a search response
+     * @throws ConstructorException if the request is invalid.
+     */
```
Comment (rubikzube) [inline constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java:544]: Comments above need a little love

File: constructorio-client/src/test/java/io/constructor/client/ConstructorIORecommendationsTest.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void createRecommendationsResponseShouldReturnAResult() throws Exception {
+        String string = Utils.getTestResource("response.recommendations.home_page_1.json");
+        RecommendationsResponse response = ConstructorIO.createRecommendationsResponse(string);
+
+        assertEquals("pod display name exists", response.getResponse().getPod().getDisplayName(), "You may also like");
+        assertEquals("pod display name exists", response.getResponse().getPod().getId(), "item_page_1");
```
Comment (rubikzube) [inline constructorio-client/src/test/java/io/constructor/client/ConstructorIORecommendationsTest.java:25]: `pod id exists`?

File: constructorio-client/src/test/java/io/constructor/client/ConstructorIORecommendationsTest.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        assertEquals("search results exist", response.getResponse().getResults().size(), 10);
```
Comment (rubikzube) [inline constructorio-client/src/test/java/io/constructor/client/ConstructorIORecommendationsTest.java:26]: `recommendation results exist`?

File: constructorio-client/src/test/java/io/constructor/client/ConstructorIORecommendationsTest.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        assertEquals("total number of results", (int)response.getResponse().getTotalNumberOfResults(), 10);
+        assertTrue("search result id exists", response.getResultId() != null);
```
Comment (rubikzube) [inline constructorio-client/src/test/java/io/constructor/client/ConstructorIORecommendationsTest.java:28]: `recommendations result id exists`?

File: constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java (hunk 1 above)
Comment (sblaurock) [reply constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java:544]: Good catch! Thanks.

**Pull request merged**